app/js/app.js

```diff
@@ -1,118 +1,135 @@
-{
+/**
+ * @class InvertedIndex
+ */
+class InvertedIndex {
   /**
-   * @class InvertedIndex
+   * Class Constructor
+   * @constructor
    */
-  class InvertedIndex {
-    /**
-     * Class Constructor
-     * @constructor
-     */
-    constructor() {
-      this.allIndices = {};
-    }
-    /**
-     * Reads the data from the file being uploaded
-     * @param  {File} file - Uploaded file to be read.
-     * @return {void}
-     */
-    static readFile(file) {
+  constructor() {
+    this.allIndices = {};
+  }
+  /**
+   * Reads the data from the file being uploaded
+   * @param  {File} file - Uploaded file to be read.
+   * @return {void}
+   */
+  static readFile(file) {
+    return new Promise ((resolve, reject) => {
       const reader = new FileReader();
-      reader.onload = () => reader.result;
+      reader.onload = () => {
+        return (objFile) => {
+          const allFileNames = [],
+          fileName = file.name,
+          fileContent = objFile.target.result;
+          try {
+            if (InvertedIndex.validateFile(fileName, fileContent)) {
+              const finalContent = JSON.parse(fileContent);
+              allFileNames.push(fileName);
+            }
+            resolve(fileContent);
+          } catch (err) {
+              reject (err);
+            }
+        };
+      };
       reader.readAsText(file);
+    }); 
+  }
+   /**
+   * Ensures all the documents in a particular file is valid
+   * @param  {Object} file
```

Comment:
Greate work here, the JSDoc ....@param should takes fileName and fileContent since validateFile method takes in two parameters (fileName and fileContent) ... 
something like...
@param {String} fileName
@param {Array} fileContent